packages/ui-components/src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,461 @@
+	};
+
+	private onDropdownChange = (event: CustomEvent<boolean>) => {
+		this.dropdownOpen = event.detail;
+		this.dropdownStateChange.emit(event.detail);
+		if (!this.isApplyButtonDisabled() && !event.detail) {
+			if (!isEmpty(this.selectedTimeOption)) {
+				this.selectedTimeState = this.selectedTimeOption;
+			} else {
+				this.selectedTimeState = {
+					key: DEFAULT_SELECTED_TIME_KEY,
+					range: getLast24HoursRange(),
+					timezone: this.getSelectedTimezone()
+				};
+			}
+
+			if (this.timePickerView !== ETimePickerView.FullView) {
+				this.timePickerView = ETimePickerView.RelativeTimePicker;
+			}
+		}
+		this.timezoneSelectionContentVisible = false;
+	};
+
+	private onSelectedRelativeTimeChange = (event: CustomEvent<ITimePickerRelativeTime>) => {
+		const timeOption = event.detail;
+		this.selectedTimeState = {
+			key: timeOption.key,
+			range: timeOption.range,
+			timezone: this.getSelectedTimezone()
+		};
+		this.calendarViewLocked = false;
+	};
+
+	private onClickSeeCustomInterval = (event: CustomEvent<string>) => {
+		const key = event.detail;
+		if (this.selectedTimeState && this.selectedTimeState.key !== CUSTOMIZE_INTERVAL_KEY) {
+			this.selectedTimeState = {
+				key: key,
+				range: [],
+				timezone: this.getSelectedTimezone()
+			};
+		}
+
+		if (!this.showCalendar) {
+			this.timePickerView = ETimePickerView.AbsoluteTimePicker;
+		} else {
+			this.calendarViewLocked = true;
+		}
+	};
+
+	private onSelectedTimezoneChange = (event: CustomEvent<ITimePickerTimezone>) => {
+		const timezone = event.detail;
+		const previousTimezone = this.selectedTimeState.timezone.name;
+		if (this.selectedTimeState.key === CUSTOMIZE_INTERVAL_KEY) {
+			this.selectedTimeState = {
+				...this.selectedTimeState,
+				range: getTimestampFromDateRange(this.selectedTimeState.range, previousTimezone, timezone.name),
+				timezone
+			};
+		} else {
+			this.selectedTimeState = {
+				...this.selectedTimeState,
+				timezone
+			};
+		}
+	};
+
+	private onClickApply = () => {
+		const payload = {
+			key: this.selectedTimeState.key,
+			range: this.selectedTimeState.range,
+			timezone: this.getSelectedTimezone()
+		};
+		this.timeRangeChange.emit(payload);
+		this.dropdownStateChange.emit(false);
+		this.dropdownOpen = false;
+		this.timezoneSelectionContentVisible = false;
+
+		if (this.timePickerView !== ETimePickerView.FullView) {
+			this.timePickerView = ETimePickerView.RelativeTimePicker;
+			this.calendarViewLocked = false;
+		}
+	};
+
+	private onClickBack = () => {
+		this.applyButtontooltipText = '';
+		this.calendarViewLocked = false;
+		if (!isEmpty(this.selectedTimeOption)) {
+			this.selectedTimeState = this.selectedTimeOption;
+		} else {
+			this.selectedTimeState = {
+				key: DEFAULT_SELECTED_TIME_KEY,
+				range: getLast24HoursRange(),
+				timezone: this.getSelectedTimezone()
+			};
+		}
+		this.timePickerView = ETimePickerView.RelativeTimePicker;
+	};
+
+	private onClickCancel = (event: CustomEvent<MouseEvent>) => {
```

Comment:
this function and `onClickBack` have a lot in common, can we reduce the replication?